app/lib/account_session.rb

```diff
@@ -75,9 +75,9 @@ def to_hash
   end
 
   def get_attributes(attribute_names)
-    local = attribute_names.select { |name| user_attributes.stored_locally? name }
-    remote = attribute_names.select { |name| !user_attributes.stored_locally?(name) && !user_attributes.cached_locally?(name) }
-    cached = attribute_names.select { |name| user_attributes.cached_locally? name }
+    local = attribute_names.select { |name| user_attributes.type(name) == "local" }
```

Comment:
Interesting! I thought type was a reserved keywork in some cases?
I thought I'd had trouble with this before via ActiveRecord, but seems to be working here.

If we can get away with it type is a great and concise name!